### DISCUSSION TITLE
What the best source navigator on Ubuntu to browse OpenROAD code

### DISCUSSION THREAD

**Discussion by titan73** [The-OpenROAD-Project/OpenROAD#6348] [Q&A] [answered]
I've used source navigator a very long time ago. Any suggestion what's the best one to use?

**Comment by QuantamHD** [comment]
Perhaps source graph meets your needs? https://sourcegraph.com/github.com/The-OpenROAD-Project/OpenROAD

**Comment by titan73** [comment]
Thanks for your link.
It does not seem to work properly though. For example in file src/sta/src/dbSdcNetwork.hh there is:
`class dbSdcNetwork : public SdcNetwork`
but I can't get to definition of SdcNetwork.

**Comment by maliberty** [comment, marked as answer]
vscode allows you to follow function calls.  Doxygen can generate html docs with hyperlinks and embedded code.

**Reply by titan73** [reply]
vscode indeed does the job. Thanks!
